core/server/common/src/main/java/alluxio/master/journal/raft/SnapshotReplicationManager.java

```diff
@@ -171,8 +174,11 @@ public SnapshotReplicationManager(RaftJournalSystem journalSystem,
       String address = String.valueOf(client.getAddress());
       SnapshotDownloader<DownloadSnapshotPRequest, DownloadSnapshotPResponse> observer =
           SnapshotDownloader.forFollower(mStorage, address);
+      Timer.Context ctx = MetricsSystem
+          .timer(MetricKey.MASTER_EMBEDDED_JOURNAL_SNAPSHOT_DOWNLOAD_TIMER.getName()).time();
       client.downloadSnapshot(observer);
       return observer.getFuture().thenApplyAsync((termIndex) -> {
+        ctx.close();
```

Comment:
Will this get closed twice, once here and once in `whenComplete`?

Reply:
Good catch, the timer got close() twice which is incorrect. Removed the close() in completeRequest. Thanks!